CGMBLEKitUI/TransmitterSettingsViewController.swift

```diff
@@ -16,15 +17,29 @@ class TransmitterSettingsViewController: UITableViewController {
 
     let cgmManager: TransmitterManager & CGMManagerUI
 
-    private var glucoseUnit: HKUnit
+    private let displayGlucoseUnitObservable: DisplayGlucoseUnitObservable
 
-    init(cgmManager: TransmitterManager & CGMManagerUI, glucoseUnit: HKUnit) {
+    private var cancellable: AnyCancellable?
+
+    private var glucoseUnit: HKUnit {
+        displayGlucoseUnitObservable.displayGlucoseUnit
+    }
+
+    init(cgmManager: TransmitterManager & CGMManagerUI, displayGlucoseUnitObservable: DisplayGlucoseUnitObservable) {
         self.cgmManager = cgmManager
-        self.glucoseUnit = glucoseUnit
+        self.displayGlucoseUnitObservable = displayGlucoseUnitObservable
 
         super.init(style: .grouped)
 
         cgmManager.addObserver(self, queue: .main)
+
+        cancellable = displayGlucoseUnitObservable.updatePublisher.sink { [weak self] in
```

Comment:
[nit] Consider above comment, plus:
```
$displayGlucoseUnitObservable
    .sink { [weak self] in self?.tableView.reloadData() }
    .store(in: &cancellables)
```
Not sure if this will work here.